src/VerifyExamples/Ast.elm

```diff
@@ -1,107 +1,117 @@
 module VerifyExamples.Ast
     exposing
         ( Ast(..)
-        , fromIntermediateAst
+        , Example
         , group
-        , isTest
         , toString
         )
 
-import VerifyExamples.IntermediateAst as IAst exposing (IntermediateAst)
-
 
 type Ast
     = Assertion String
     | Expectation String
     | Import String
-    | LocalFunction String String
+    | Function String String
     | Type String
 
 
-isTest : Ast -> Bool
-isTest x =
-    case x of
-        Assertion _ ->
-            True
+type alias Groupped =
+    { functions : List Ast
+    , imports : List Ast
+    , types : List Ast
```

Comment:
It seems to me that having these stored as AST will lead to boilerplate and cases that are impossible to reach. How about turning the AST for each into a dedicated token at this point?

Reply:
totally. that's way better